pkg/features/kube_features.go

```diff
@@ -1009,9 +1009,9 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	genericfeatures.StorageVersionHash: {Default: true, PreRelease: featuregate.Beta},
 
-	genericfeatures.StrictCostEnforcementForVAP: {Default: false, PreRelease: featuregate.Beta},
+	genericfeatures.StrictCostEnforcementForVAP: {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```

Comment:
This should be updated to use versioned feature gates. @jpbetz can we merge  https://github.com/kubernetes/kubernetes/pull/127200 first so we won't have rebase conflicts?

Reply:
Looks like 127200 also blocked by the merging of other PR currently. Is that possible to merge this PR first and I'll have a followup PR to add the deleted tests back?

Reply:
I'll defer to @jpbetz on this, though 127200 shouldn't be blocked and just needs approval.

Reply:
If you're looking to merge this PR first, I'd probably recommend keeping the tests and skipping them with a TODO rather than deleting and re-adding.

Reply:
I added a new commit which putting back the tests with TODO comments. Will see which one merged first. :)